File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson03Inheritance/Exercise/pr06_animals/Tomcat.java
package Lesson03Inheritance.Exercise.pr06_animals;

public class Tomcat extends Cat {

    private static final String SOUND = "Give me one million b***h";

    public Tomcat(String name, int age, String gender) {
        super(name, age, gender);
    }

    @Override
    protected void setGender(String gender) {
        if (!MessageConstants.MALE.equalsIgnoreCase(gender)) {
            throw new IllegalArgumentException(MessageConstants.INVALID_INPUT);
        }
        super.setGender(gender);
    }

    @Override
    public String produceSound() {
        return SOUND;
    }
}

File: C#-OOP-Basics/Problems Course Instance 2018/Inheritance/Animals/Cats/Tomcat.cs
public class Tomcat : Cat
{
    public Tomcat(string name, string age) : base(name, age)
    {
        base.Gender = "Male";
    }

    public Tomcat(string name, string age, string gender) : this(name, age)
    {
    }

    public override string ProduceSound()
    {
        return "MEOW";
    }
}